cfme/tests/infrastructure/test_host_role_type.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+from utils import testgen
+from cfme.web_ui import Quadicon
+from cfme.web_ui import InfoBlock
+from cfme.fixtures import pytest_selenium as sel
+import re
+
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack-infra'],
+                                         scope='module')
+
+
+ROLES = ['Compute', 'Controller', 'BlockStorage', 'SwiftStorage',
+         'CephStorage']
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_host_role_type(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    result = False
+    for quad in my_quads:
+        role_name = str(quad.name)
+        role_name = re.search(r'\((\w+)\)', role_name).group(1)
+        if role_name in ROLES:
```

Comment:
This test will assert that at least one role is found, but if any of the others are not found, then the assert will still be true, is this desired?